File: iLihuxAutoLayoutInOC/iLihuxAutoLayoutInOC/LHTableViewCell.m
//
//  LHTableViewCell.m
//  iLihuxAutoLayoutInOC
//
//  Created by lihux on 14/11/30.
//  Copyright (c) 2014年 lihux. All rights reserved.
//

#import "LHTableViewCell.h"

@implementation LHTableViewCell

- (IBAction)didTapOnCancelButton:(id)sender
{
    if (self.delegate && [self.delegate respondsToSelector:@selector(cancelButtonTapped)]) {
        [self.delegate cancelButtonTapped];
    }
}

@end

File: iLihuxAutoLayoutInSwift/iLihuxAutoLayoutInSwift/LHTableViewCell.swift
//
//  LHTableViewCell.swift
//  iLihuxAutoLayoutInSwift
//
//  Created by lihux on 14/11/30.
//  Copyright (c) 2014年 lihux. All rights reserved.
//

import UIKit

@objc protocol UITableViewCellDelegate: NSObjectProtocol{
    optional func cancelButton() -> Void
}

class LHTableViewCell: UITableViewCell {

    weak var delegate: UITableViewCellDelegate?

    @IBAction func didTapOnCancelButton(sender: AnyObject)
    {
        if let owner = self.delegate {
            if owner.respondsToSelector(Selector("cancelButton")) {
                owner.cancelButton!()
            }
        }
    }

}
